simulation/simple_sensor_simulator/src/vehicle_simulation/ego_entity_simulation.cpp

```diff
@@ -297,6 +311,90 @@ void EgoEntitySimulation::update(
   updatePreviousValues();
 }
 
+auto EgoEntitySimulation::getMatchedLaneletPoseFromEntityStatus(
+  const traffic_simulator_msgs::msg::EntityStatus & status, const double entity_width) const
+  -> std::optional<traffic_simulator_msgs::msg::LaneletPose>
+{
+  // the lanelet matching algorithm should be equivalent to the one used in EgoEntity::setMapPose
+  const auto unique_route_lanelets =
+    traffic_simulator::helper::getUniqueValues(autoware->getRouteLanelets());
+  const auto matching_length = [entity_width] { return entity_width * 0.5 + 1.0; }();
+
+  std::optional<traffic_simulator_msgs::msg::LaneletPose> lanelet_pose;
+
+  if (unique_route_lanelets.empty()) {
+    lanelet_pose =
+      hdmap_utils_ptr_->toLaneletPose(status.pose, status.bounding_box, false, matching_length);
+  } else {
+    lanelet_pose =
+      hdmap_utils_ptr_->toLaneletPose(status.pose, unique_route_lanelets, matching_length);
+    if (!lanelet_pose) {
+      lanelet_pose =
+        hdmap_utils_ptr_->toLaneletPose(status.pose, status.bounding_box, false, matching_length);
+    }
+  }
+  return lanelet_pose;
+}
+
+auto EgoEntitySimulation::calculateEgoPitch() const -> double
+{
+  // calculate prev/next point of lanelet centerline nearest to ego pose.
+  auto ego_lanelet = getMatchedLaneletPoseFromEntityStatus(
+    status_, std::max(
+               vehicle_parameters.axles.front_axle.track_width,
+               vehicle_parameters.axles.rear_axle.track_width));
+  if (not ego_lanelet) {
+    return 0.0;
+  }
+
+  auto centerline_points = hdmap_utils_ptr_->getCenterPoints(ego_lanelet.value().lanelet_id);
+
+  // copied from motion_util::findNearestSegmentIndex
```

Comment:
```suggestion
  // @note Copied from motion_util::findNearestSegmentIndex
```

Reply:
Please show the permalink of the Autoware code you referenced using the @sa command in Doxygen.